pkgs/development/python-modules/gepetto-viewer/default.nix

```diff
@@ -9,65 +9,83 @@
   doxygen,
   jrl-cmakemodules,
   libsForQt5,
+  makeWrapper,
   openscenegraph,
   osgqt,
   pkg-config,
   python-qt,
   qgv,
 }:
-buildPythonPackage rec {
-  pname = "gepetto-viewer";
-  version = "5.1.0";
-  pyproject = false; # CMake
+let
+  gepetto-viewer = buildPythonPackage rec {
+    pname = "gepetto-viewer";
+    version = "5.1.0";
+    pyproject = false; # CMake
 
-  src = fetchFromGitHub {
-    owner = "gepetto";
-    repo = "gepetto-viewer";
-    rev = "v${version}";
-    hash = "sha256-A2J3HidG+OHJO8LpLiOEvORxDtViTdeVD85AmKkkOg8=";
-  };
+    src = fetchFromGitHub {
+      owner = "gepetto";
+      repo = "gepetto-viewer";
+      rev = "v${version}";
+      hash = "sha256-A2J3HidG+OHJO8LpLiOEvORxDtViTdeVD85AmKkkOg8=";
+    };
+
+    cmakeFlags = [
+      (lib.cmakeBool "BUILD_PY_QCUSTOM_PLOT" (!stdenv.isDarwin))
+      (lib.cmakeBool "BUILD_PY_QGV" (!stdenv.isDarwin))
+    ];
 
-  cmakeFlags = [
-    (lib.cmakeBool "BUILD_PY_QCUSTOM_PLOT" (!stdenv.isDarwin))
-    (lib.cmakeBool "BUILD_PY_QGV" (!stdenv.isDarwin))
-  ];
+    outputs = [
+      "out"
+      "doc"
+    ];
 
-  outputs = [
-    "out"
-    "doc"
-  ];
+    buildInputs = [
+      boost
+      python-qt
+      libsForQt5.qtbase
+    ];
 
-  buildInputs = [
-    boost
-    python-qt
-    libsForQt5.qtbase
-  ];
+    nativeBuildInputs = [
+      cmake
+      doxygen
+      libsForQt5.wrapQtAppsHook
+      pkg-config
+    ] ++ lib.optionals (stdenv.isDarwin && stdenv.isAarch64) [ darwin.autoSignDarwinBinariesHook ];
 
-  nativeBuildInputs = [
-    cmake
-    doxygen
-    libsForQt5.wrapQtAppsHook
-    pkg-config
-  ] ++ lib.optionals (stdenv.isDarwin && stdenv.isAarch64) [ darwin.autoSignDarwinBinariesHook ];
+    propagatedBuildInputs = [
+      jrl-cmakemodules
+      openscenegraph
+      osgqt
+      qgv
+    ];
 
-  propagatedBuildInputs = [
-    jrl-cmakemodules
-    openscenegraph
-    osgqt
-    qgv
-  ];
+    doCheck = true;
 
-  doCheck = true;
+    # wrapQtAppsHook uses isMachO, which fails to detect binaries without this
+    # ref. https://github.com/NixOS/nixpkgs/pull/138334
+    preFixup = lib.optionalString stdenv.isDarwin "export LC_ALL=C";
 
-  # wrapQtAppsHook uses isMachO, which fails to detect binaries without this
-  # ref. https://github.com/NixOS/nixpkgs/pull/138334
-  preFixup = lib.optionalString stdenv.isDarwin "export LC_ALL=C";
+    passthru.withPlugins =
+      plugins:
+      stdenv.mkDerivation {
```

Comment:
Why not use runCommand?

Reply:
The OP

Reply:
I didn't knew how to use it, but yes, it seems way better, thanks !